cranelift-wasm/src/module_translator.rs

```diff
@@ -137,6 +137,16 @@ pub fn translate_module<'data>(
     if let SectionCode::Data = section.code {
         let data = section.get_data_section_reader()?;
         parse_data_section(data, environ)?;
+
+        if reader.eof() {
+            return Ok(());
+        }
+        section = reader.read()?;
+    }
+
+    if let SectionCode::Custom { name: "name", .. } = section.code {
+        let names = section.get_name_section_reader()?;
+        parse_name_section(names, environ)?;
```

Comment:
The [wasm specification](https://webassembly.github.io/spec/core/appendix/custom.html#name-section) mentions that the name section should come after the data section, so this block is placed here.